crates/iota-json-rpc/src/governance_api.rs

```diff
@@ -802,4 +832,39 @@ mod tests {
             assert!(apy.apy < 0.25)
         }
     }
+
+    #[test]
+    fn test_backfill_rates_empty() {
+        let rates = vec![];
+        assert_eq!(backfill_rates(rates), vec![]);
+    }
+
+    #[test]
+    fn test_backfill_rates_no_gaps() {
+        let rate1 = PoolTokenExchangeRate::new_for_testing(100, 100);
+        let rate2 = PoolTokenExchangeRate::new_for_testing(200, 220);
+        let rate3 = PoolTokenExchangeRate::new_for_testing(300, 330);
+        let rates = vec![(2, rate2.clone()), (3, rate3.clone()), (1, rate1.clone())];
+
+        let expected: Vec<(u64, PoolTokenExchangeRate)> =
+            vec![(3, rate3.clone()), (2, rate2), (1, rate1)];
```

Comment:
Please make this multiline, as it is easier to read.

Reply:
I agree it would be nicer to have
```rust
        let rates = vec![(2, rate2.clone()), 
                                  (3, rate3.clone()), 
                                  (1, rate1.clone())];
        let expected: Vec<(u64, PoolTokenExchangeRate)> =
            vec![(3, rate3.clone()), 
                 (2, rate2), 
                 (1, rate1)];
 ```
 but looks like the `cargo +nightly fmt` will force them to be one line.